### ISSUE TITLE
Cannot use a DeferredList to retrieve several POP3 messages

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1988]
|<img alt="kendrew's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| kendrew reported|
|-|-|
|Trac ID|trac#1988|
|Type|defect|
|Created|2006-08-07 04:11:18Z|

<details><summary>Searchable metadata</summary>

```
trac-id__1988 1988
type__defect defect
reporter__kendrew kendrew
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1154923878000000 1154923878000000
changetime__1155508472000000 1155508472000000
version__None None
owner__ 
cc__exarkun cc__kendrew
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set owner to kendrew|
|-|-|

I'm sorry, I don't know what this ticket means.  You will need to provide more details.  Ideally, you should attach a failing unit test to the ticket.  At least attach a simple program I can run which you believe should work but which fails in some way.  Please explain what the desired and actual behaviors are.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

I am guessing this ticket means:


  p.retrieve(a)
  p.retrieve(b)
  p.retrieve(c)

doesn't work right, since it doesn't.  Fix ready for review in pop3-concurrent-retrieve-1988

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix set owner to @radix<br>@radix set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_4"></a>|@radix set owner to @exarkun<br>@radix set status to `new`|
|-|-|

It looks like every function in _blockedQueue *needs* to reset self._blockedQueue to a list, given that there's code like

```
self._blockedQueue = None
f()
self._blockedQueue.extend(blah)
```

If my understanding is correct, I suggest changing the line "f()" to "f() # Must reset self._blockedQueue"


Otherwise +1.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [17820]) Merge pop3-concurrent-retrieve-1988

Author: exarkun
Reviewer: radix
Fixes [#1988](https://github.com/twisted/twisted/issues/1988)
Fixes [#11176](https://github.com/twisted/twisted/issues/11176)

This fixes the POP3 client to correctly unblock itself after a command
completes successfully and there is more than one other command pending.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
